test/object-store/util/baas_admin_api.hpp

```diff
@@ -158,16 +158,23 @@ struct AppCreateConfig {
         bool run_reset_function;
     };
 
-    struct FLXSyncRole {
+    struct ServiceRoleDocumentFilters {
+        nlohmann::json read;
+        nlohmann::json write;
+    };
+
+    struct ServiceRole {
         std::string name;
         nlohmann::json apply_when = nlohmann::json::object();
+        ServiceRoleDocumentFilters document_filters;
```

Comment:
So what exactly does this do/mean? What's the difference between `insert_filter` and `write`?

Reply:
for the purposes of flexible sync, document_filters.read/document_filters.write replace how the old "read"/"write" in the flexible sync role was used. You might be wondering why we couldn't just map the top-level "read"/"write" in a flex sync role to the top-level "read"/"write" in the service roles.... which is a long story. If you're curious, see the [scope](https://docs.google.com/document/d/1_VUXtnZT2xXlc5Hh4FzYrL3LbgHl6ZsgvsG7lBOdmBM/edit#heading=h.3wkklyocpznh) for more details, but the tl;dr is that top-level "read"/"write" in service roles for the non-sync services (GraphQL, DataAPI, etc.) are used in a slightly different way than how the top-level "read"/"write" flexible sync role permissions were being used.   

service roles support permissions for restricting when someone has access to insert/delete a MDB document. that is what the `insert_filter` and `delete_filter` fields here are supposed to represent (side note - I was originally going to call these fields `insert`/`delete` in the struct, but `delete` is a reserved keyword). As these permissions formerly had not existed in the old flexible sync role structure, these are permissions that are now supported in flexible sync as well (a client can only create/erase objects if they have the corresponding permission)

imo, the easiest way to think about top-level `read`/`write` in a service role is in the context of field-level permissions: as "read all fields"/"write all fields" permissions

Reply:
the [spec](https://docs.google.com/document/d/1uGgZiYrSidKswCkNSAwVvJFVFhUiq5YJmOoHlWfZX44/edit?pli=1#heading=h.heh2001b7qok) also has some more details on these fields

Reply:
Do we need all these fields? It looks like we set most of them to true everywhere we use this struct.

Reply:
I think from the compensating writes tests here, it looks like we were setting the old top-level "write" flex sync role field to cause invalid writes on the server; but as far as I can tell, it looks like the other tests don't leverage permissions at all (ie, just use open permissions). Happy to remove some of the code here if you think including things like "insert"/"delete"/"read"/"write" are noisy, but I included them in case we ever wanted to leverage them in the future

Reply:
I think we should at least add a comment explaining why/when you should set the values and maybe give them  sensible defaults are. Like if we add a comment explaining "set this to false if you want to disallow writes" and have the default value of true - I think that'd be helpful.

Reply:
Added in some comments explaining what these different permissions do, lmk what you think!